spec/jobs/import_titmc_stock_units_spec.rb

```diff
@@ -82,4 +82,41 @@
       subject
     end
   end
+
+  describe '[integration] Failure' do
+    let(:stock_unit) { create :stock_unit_wildcard, status: 'PENDING' }
+    let(:id) { stock_unit.id }
+
+    before do
+      allow(DataSource::Stock::TribunalInstance::Unit::Operation::LoadTransmission)
+        .to receive(:call)
+        .and_wrap_original {
+          # TODO: create something related to TITMC
+          create :stock_unit, status: 'GHOST'
+          trb_result_failure
+        }
+    end
+
+    it 'set status to ERROR' do
+      subject
+      stock_unit.reload
+
+      expect(stock_unit.status).to eq 'ERROR'
+    end
+
+    it 'rollbacks database' do
+      subject
+      ghost = StockUnit.where(status: 'GHOST')
+      expect(ghost).to be_empty
+    end
+
+    it 'operation has an error' do
```

Comment:
Je ne comprends pas ce qui est testé ici.

Reply:
et si je renomme ainsi : `it 'calls an operation that fails'` ?

Reply:
En fait je comprends pas ce qui est testé : si je lis le fichier de specs, en reprenant depuis le groupe parent ça donne "when LoadTransmission fails *it calls an operation that fails*" ? Quelle opération ?

Reply:
ici on est dans le Job `ImportTitmcStockUnit` qui appelle l'opération `DataSource::Stock::TribunalInstance::Unit::Operation::Load` qui elle même appelle `LoadTransmission.call` qui échoue

C'est une sorte de test d'intégration, je m'assure que l'erreur est bien remonté au top level càd le job.
Après avoir explicité ce test je pense qu'on peux s'en passer

Reply:
Ok je comprends ^^ je vois pas vraiment d'intérêt à ce test : c'est pas vraiment de la responsabilité du job de tester les cas de tests de *Load*. Son boulot au job c'est d'appeler *Load* et de gérer les cas de succès et d'échecs, pourquoi *Load* échoue il s'en moque. Là c'est un coup à casser le tests si l'opération Load évolue et que la dépendance à *LoadTransmission* change ou est remplacée, alors quand vrai le job fait toujours son taf et marche correctement.

Par contre, tester que quand *LoadTransmission* échoue alors *Load* échoue ça oui il faut s'en assurer.